modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/src/main/java/com/liferay/headless/commerce/delivery/cart/internal/resource/v1_0/ShippingMethodResourceImpl.java

```diff
@@ -0,0 +1,182 @@
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.headless.commerce.delivery.cart.internal.resource.v1_0;
+
+import com.liferay.commerce.context.CommerceContext;
+import com.liferay.commerce.context.CommerceContextFactory;
+import com.liferay.commerce.currency.util.CommercePriceFormatter;
+import com.liferay.commerce.model.CommerceAddress;
+import com.liferay.commerce.model.CommerceOrder;
+import com.liferay.commerce.model.CommerceShippingEngine;
+import com.liferay.commerce.model.CommerceShippingMethod;
+import com.liferay.commerce.model.CommerceShippingOption;
+import com.liferay.commerce.product.model.CommerceChannel;
+import com.liferay.commerce.product.service.CommerceChannelLocalService;
+import com.liferay.commerce.service.CommerceOrderService;
+import com.liferay.commerce.service.CommerceShippingMethodService;
+import com.liferay.commerce.util.CommerceShippingEngineRegistry;
+import com.liferay.commerce.util.comparator.CommerceShippingOptionLabelComparator;
+import com.liferay.headless.commerce.delivery.cart.dto.v1_0.ShippingMethod;
+import com.liferay.headless.commerce.delivery.cart.dto.v1_0.ShippingOption;
+import com.liferay.headless.commerce.delivery.cart.resource.v1_0.ShippingMethodResource;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.vulcan.pagination.Page;
+
+import java.math.BigDecimal;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+
+/**
+ * @author Andrea Sbarra
+ */
+@Component(
+	properties = "OSGI-INF/liferay/rest/v1_0/shipping-method.properties",
+	scope = ServiceScope.PROTOTYPE, service = ShippingMethodResource.class
+)
+public class ShippingMethodResourceImpl extends BaseShippingMethodResourceImpl {
+
+	@Override
+	public Page<ShippingMethod> getCartShippingMethodsPage(Long cartId)
+		throws Exception {
+
+		CommerceOrder commerceOrder = _commerceOrderService.getCommerceOrder(
+			cartId);
+
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.getCommerceChannelByOrderGroupId(
+				commerceOrder.getGroupId());
+
+		CommerceAddress shippingAddress = commerceOrder.getShippingAddress();
+
+		if (shippingAddress != null) {
+			return Page.of(
+				transform(
+					_commerceShippingMethodService.getCommerceShippingMethods(
+						commerceChannel.getGroupId(),
+						shippingAddress.getCommerceCountryId(), true),
+					shippingMethod -> _toShippingMethod(shippingMethod,
+					commerceChannel, commerceOrder)));
+		}
```

Comment:
We should rename shippingAddress to shippingCommerceAddress. We can also move commerceChannel inside the if statement since its only being used in there.
```suggestion
		CommerceAddress shippingCommerceAddress = commerceOrder.getShippingAddress();

		if (shippingCommerceAddress != null) {
			CommerceChannel commerceChannel =
				_commerceChannelLocalService.getCommerceChannelByOrderGroupId(
					commerceOrder.getGroupId());

			return Page.of(
				transform(
					_commerceShippingMethodService.getCommerceShippingMethods(
						commerceChannel.getGroupId(),
						shippingCommerceAddress.getCommerceCountryId(), true),
					shippingMethod -> _toShippingMethod(shippingMethod,
					commerceChannel, commerceOrder)));
		}
```